javascript/javascript1/week3/smartApps.js

```diff
@@ -0,0 +1,85 @@
+const activities = [];
+let date = new Date();
+let day = date.getDate();
+let months = date.getMonth() + 1;
+let year = date.getFullYear();
+
+function getSum(arr) {
+  let counter = arr.reduce((a, b) => {
+    return a + b
+  })
+  return counter
+}
+
+function getTime(time) {
+  let hours = Math.floor(time / 60)
+  let minuts = time % 60
+  const timeArr = [time, hours, minuts]
+  return timeArr
+}
+
+
+function addActivity(activity, duration) {
+  if (typeof activity != 'string' || typeof duration != 'number')
+    throw new Error('Illegal Argument Exception');
+
+  let date = `${day}-${months}-${year}`
+  const obj = { 'date': `${date}`, 'activity': `${activity}`, 'duration': duration }
+  activities.push(obj)
```

Comment:
Generally you should avoid side effects when possible. In this case it would be better to take the array as a parameter to the function, or perhaps rename the function to getActivity and have it just return the constructed activity, which can then be added to the array by the caller. Each effect outside the closure of the function increases complexity, so avoid when possible.